microsoftgraph/microsoft-graph-explorer-v4
# Task: Implement lazy loading

## Overview
Implements #2228 

### Demo

Optional. Screenshots, `curl` examples, etc.
<img width="313" alt="image" src="https://user-images.githubusercontent.com/45680252/203726121-f3f848f3-f679-41fa-be86-71ee172fc38b.png">
Chunks before lazy loading

![image](https://user-images.githubusercontent.com/45680252/229475005-5cdcdfbb-cf30-41f3-a177-c123bfe0b39c.png)

Chunks after lazy loading


### Notes
Currently, Graph Explorer is loaded in a single bundle. With the app growing, this bundle is also increasing in size. The current bundle has a size a little over 1MB. This has an impact on the app load time as the whole bundle has to be loaded first for the app to become usable

Lazy loading splits the original bundle into smaller chunks so that what is needed on first-load is what is loaded initially. The lazily loaded components will be in their own chunks and will be loaded on demand

## Testing Instructions

* How to test this PR
Load the app on your browser
Click 'Request headers' tab
Notice that it is loaded on demand
Click on other components (Modify Permissions, Access token, Response headers, Resources etc.)
